### PR TITLE
Hatchery queue QoL improvements

### PR DESCRIPTION
**Pull request by CypherX**
Adds several QoL improvements to the hatchery queue:

- Display amount of pokemon currently in queue and total amount of queue slots available.
- Add a Clear button to clear the entire queue.
- Setting to limit the queue size to any of the previously obtained regional increments (4, 8, 16, 32, 64, 96) or disable the queue entirely.
    - From: #1358
    - Queue size setting will not be displayed if the queue is not unlocked.
    - Switching to a smaller queue size will remove any extra pokemon from the queue.

image
image

### CODE REVIEW

File: src/scripts/breeding/Breeding.ts [hunk 1]
```diff
@@ -456,4 +463,35 @@ class Breeding implements Feature {
         }
     }
 
+    private usableQueueSlots = ko.pureComputed(() => {
```
Comment (Jaaslet) [inline src/scripts/breeding/Breeding.ts:466]: It should be public, now that it's used in the html. Works without, but private might make people think that it's not used elsewhere

**Review by Jaaslet — CHANGES_REQUESTED** [review]
Looks good. I have some small changes:
- I think the clear button needs a confirmation dialog. It's easy to hit, and can ruin a lot of clicks.
- The queue size should be moved to the settings window. It will not be changed by many, and the people who wants to use it, will only change it once, so no need to use UI space in the hatchery

**Comment by CypherX** [comment]
Thanks for the feedback.

- Added confirmation to the Clear queue button.
- Made usableQueueSlots public.
- Moved the queue size setting to the regular settings window under the Other tab. Let me know if there's a better way to handle the setting, the options being dynamic based on player progression make it a bit different from the other settings that have static options.

**Comment by RedSparr0w** [comment]
The current way the setting is implemented, it won't update when the player travels to a new region.
You could bind to update the settings whenever the `queueSlots` value is updated.
Or just allow the player to enter any number instead of having a dropdown (probably add a bind to limit to max amount on change).

**Comment by CypherX** [comment]
Alright, changed the setting to use a textbox instead of dropdown to allow the player to enter any (valid) number. My only reservation is using -1 to denote 'no limit' (also the default value) may be confusing to some players. I've included a tooltip to hopefully help with any confusion. I can't see many people wanting to use this setting anyway, lol.

**Review by Jaaslet — APPROVED** [review]
Looks good. Thanks for the PR

**Pull request merged**